docs/taskruns.md

```diff
@@ -4,92 +4,76 @@ linkTitle: "TaskRuns"
+
+# Overview
+
+A `TaskRun` allows you to instantiate and execute a [`Task`](tasks.md) on-cluster. A `Task` specifies one or more
+`Steps` that execute container images that each perform a specific piece of build work. A `TaskRun` executes the
+`Steps` in the `Task` in the order they are specified until all `Steps` have executed successfully or a failure occurs.
+
+## Configuring a `TaskRun`
+
+A `TaskRun` definition supports the following fields:
 
 - Required:
   - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
     `tekton.dev/v1beta1`.
-  - [`kind`][kubernetes-overview] - Specify the `TaskRun` resource object.
-  - [`metadata`][kubernetes-overview] - Specifies data to uniquely identify the
-    `TaskRun` resource object, for example a `name`.
-  - [`spec`][kubernetes-overview] - Specifies the configuration information for
-    your `TaskRun` resource object.
-    - [`taskRef` or `taskSpec`](#specifying-a-task) - Specifies the details of
-      the [`Task`](tasks.md) you want to run
+  - [`kind`][kubernetes-overview] - Identifies this resource object as a `TaskRun` object.
+  - [`metadata`][kubernetes-overview] - Specifies metadata that uniquely identifies the
+    `TaskRun`, such as a `name`.
+  - [`spec`][kubernetes-overview] - Specifies the configuration for the `TaskRun`.
+    - [`taskRef` or `taskSpec`](#specifying-the-target-task) - Specifies the `Tasks` that the
+    `TaskRun` will execute.
 - Optional:
-
-  - [`serviceAccountName`](#service-account) - Specifies a `ServiceAccount` resource
-    object that enables your build to run with the defined authentication
-    information. When a `ServiceAccount` isn't specified, the `default-service-account`
-    specified in the configmap `config-defaults` will be applied.
-  - [`params`](#parameters) - Specifies parameters values
-  - [`resources`](#providing-resources) - Specifies `PipelineResource` values
-    - [`inputs`] - Specifies input resources
-    - [`outputs`] - Specifies output resources
-  - [`timeout`] - Specifies timeout after which the `TaskRun` will fail. If the value of
-    `timeout` is empty, the default timeout will be applied. If the value is set to 0,
-    there is no timeout. You can also follow the instruction [here](#Configuring-default-timeout)
-    to configure the default timeout.
-  - [`podTemplate`](#pod-template) - Specifies a [pod template](./podtemplates.md) that will be used as the basis for the `Task` pod.
-  - [`workspaces`](#workspaces) - Specify the actual volumes to use for the
-    [workspaces](workspaces.md#declaring-workspaces-in-tasks) declared by a `Task`
+  - [`serviceAccountName`](#specifying-serviceaccount-credentials) - Specifies a `ServiceAccount`
+    object that provides custom credentials for executing the `TaskRun`.
+  - [`params`](#specifying-parameters) - Specifies configuration parameters.
+  - [`resources`](#specifying-resources) - Specifies `PipelineResource` values.
```

Comment:
We don't yet know here which or how many parameters or `PipelineResource` values these fields specify so "the" doesn't work here. It could be any number or combination of parameters. Omitting the article implies "some" which I think is appropriate here.

Reply:
You're right, though I think it works better if we get more specific about the parameters and values. Reworded to "specifies the desired execution parameters for the `Task`".